lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -423,20 +443,18 @@ SingleExitLoopTransformer::computeEscapingValues() const {
     }
   }
 
-  // Do not eliminate undefs unless requested for.
-  if (!TFNoUndefsInSESE)  return result;
+  return getPreheaderSubstMap(escapingValues);
+}
 
-  // Build the equivalence classes induced by argument passing.
-  llvm::EquivalenceClasses<SILValue> equivalentValues;
-  for (auto &bb : *currentFn) {
-    for (auto arg : bb.getArguments()) {
-      SmallVector<SILValue, 8> incomingValues;
-      arg->getIncomingValues(incomingValues);
-      for (SILValue incomingValue : incomingValues) {
-        equivalentValues.unionSets(arg, incomingValue);
-      }
-    }
+llvm::DenseMap<SILValue, SILValue>
+SingleExitLoopTransformer::getPreheaderSubstMap(
+    const SmallPtrSetImpl<SILValue> &values) const {
+  llvm::DenseMap<SILValue, SILValue> result;
+  for (const SILValue value : values) {
+    result[value] = getUndef(value->getType());
   }
+  // Do not eliminate undefs unless requested for.
+  if (!TFNoUndefsInSESE)  return result;
```

Comment:
Seems to be a double-width space before `return`?

Reply:
done.